File: Domain/Recipe/Recipe.cs
using Domain.User;

namespace Domain.Recipes {

    public class Recipe {

        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public DateTime Date { get; set; }
        public string Cuisine { get; set; }
        public string CookTime { get; set; }

        public string Image { get; set; }

        public string AuthorId { get; set; }
        public AppUser Author { get; set; }

        public List<RecipeIngredient> RecipeIngredients { get; set; }
        public List<Instruction> Instructions { get; set; }
        public ICollection<Comment> Comments { get; set; }

        public Recipe() {
            RecipeIngredients = new List<RecipeIngredient>();
            Instructions = new List<Instruction>();
            Comments = new List<Comment>();
        }
    }
}

File: client/src/app/models/recipe.ts
import { Profile } from './profile';

// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
export interface Recipe {
	id: string;
	name: string;
	description: string;
	date: Date | null;
	category: string;
	cookTime: string;
	cuisine: string;
	author?: Profile;
	authorUsername?: string;
	isAuthor?: boolean;
	authorId?: string;
	image: string;
	recipeIngredients: RecipeIngredient[];
	instructions: RecipeInstruction[];
}

export interface RecipeInstruction {
	step: number;
	text: string;
}

export interface RecipeIngredient {
	amount: number;
	unit: string;
	ingredient: {
		name: string;
	};
}

export class RecipeFormValues {
	id?: string = undefined;
	name: string = '';
	category: string = '';
	description: string = '';
	date: Date | null = null;
	cookTime: string = '';
	cuisine: string = '';
	image: string = '';
	recipeIngredients: RecipeIngredient[] = [];
	instructions: RecipeInstruction[] = [];

	constructor(recipe?: RecipeFormValues) {
		if (recipe) {
			this.id = recipe.id;
			this.name = recipe.name;
			this.category = recipe.category;
			this.description = recipe.description;
			this.date = recipe.date;
			this.cookTime = recipe.cookTime;
			this.image = recipe.image;
			this.cuisine = recipe.cuisine;
			this.recipeIngredients = recipe.recipeIngredients;
			this.instructions = recipe.instructions;
		}
	}
}

// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
export class Recipe implements Recipe {
	constructor(init?: RecipeFormValues) {
		Object.assign(this, init);
	}
}
